### PR TITLE
Refactor code quality issues

### PR DESCRIPTION
**Pull request by akshgpt7**
## Description

Hi, I ran DeepSource analysis on my fork of the repo, and found some interesting [code quality and performance issues](https://deepsource.io/gh/akshgpt7/pretix/) to consider.
This PR fixes a few of the issues detected.

## Summary of changes
- Use literals instead of function calls to assign data structure.
- Refactor unnecessary use of list comprehension.
- Merge `isinstance` calls.
- Use `if` statements to improve readability.
- Remove calling `len()` for comparison.

### CODE REVIEW

File: src/pretix/plugins/checkinlists/exporters.py [hunk 1]
```diff
@@ -159,7 +159,7 @@ def _get_queryset(self, cl, form_data):
             ), self.event.timezone)
             qs = qs.filter(subevent__date_from__lt=dt)
 
-        o = tuple()
+        o = ()
```
Comment (raphaelm) [inline src/pretix/plugins/checkinlists/exporters.py:162]: Why do you somethings think adding tuple is good (see above) and sometimes that removing it is good?

File: .deepsource.toml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+version = 1
```
Comment (raphaelm) [inline .deepsource.toml:1]: Please remove this file from the PR, we don't plan on using your tool regularly and the repo has enough config files already :)

File: src/pretix/plugins/checkinlists/exporters.py (hunk 1 above)
Comment (akshgpt7) [reply src/pretix/plugins/checkinlists/exporters.py:162]: I don't get you, I have removed all unnecessary `tuple()` calls.

File: .deepsource.toml (hunk 2 above)
Comment (akshgpt7) [reply .deepsource.toml:1]: Sure :)

**Comment by akshgpt7** [comment]
I can't figure out why the tests are failing. Any pointers to fix them would be appreciated :)

File: src/pretix/plugins/checkinlists/exporters.py (hunk 1 above)
Comment (raphaelm) [reply src/pretix/plugins/checkinlists/exporters.py:162]: Sorry, misread the diff. Disregard

File: src/pretix/plugins/sendmail/tasks.py [hunk 3]
```diff
@@ -36,9 +36,8 @@ def send_mails(event: Event, user: int, subject: dict, message: dict, orders: li
 
                 if filter_checkins:
                     checkins = list(p.checkins.all())
-                    allowed = (
-                        (not_checked_in and not checkins)
-                        or (any(c.list_id in checkin_lists for c in checkins))
+                    allowed = not checkins if not_checked_in else (
+                        any(c.list_id in checkin_lists for c in checkins)
```
Comment (raphaelm) [inline src/pretix/plugins/sendmail/tasks.py:40]: The failing tests seem to be related to this one. Note that previously ``allowed`` was ``True`` when ``not_checked_in = True, checkins = True, any(…) = True``) and now it no longer is.

File: src/pretix/control/views/typeahead.py [hunk 4]
```diff
@@ -632,10 +632,9 @@ def meta_values(request):
 
         defaults = defaults.filter(organizer_id=organizer.pk)
         matches = matches.filter(event__organizer_id=organizer.pk)
-        all_access = (
-            (request and request.user.has_active_staff_session(request.session.session_key))
-            or request.user.teams.filter(all_events=True, organizer=organizer).exists()
-        )
+        all_access = request.user.has_active_staff_session(
+            request.session.session_key) if request else request.user.teams.filter(
+            all_events=True, organizer=organizer).exists()
```
Comment (raphaelm) [inline src/pretix/control/views/typeahead.py:637]: After we figured out the other one caused a problem I'd very much prefer to not have this modification as well. It's not the same thing.

File: src/pretix/base/services/orders.py [hunk 5]
```diff
@@ -2089,7 +2089,7 @@ def _try_auto_refund(order, manual_refund=False, allow_partial=False, source=Ord
     else:
         proposals = order.propose_auto_refunds(refund_amount)
         can_auto_refund_sum = sum(proposals.values())
-        can_auto_refund = (allow_partial and can_auto_refund_sum) or can_auto_refund_sum == refund_amount
+        can_auto_refund = can_auto_refund_sum if allow_partial else can_auto_refund_sum == refund_amount
```
Comment (raphaelm) [inline src/pretix/base/services/orders.py:2092]: After we figured out the other one caused a problem I'd very much prefer to not have this modification as well. It's not the same thing.

**Comment by akshgpt7** [comment]
Thanks @raphaelm, I've reverted those changes.

**Review by raphaelm — APPROVED** [review]

**Pull request merged**